std/os/file.zig

```diff
@@ -85,6 +85,14 @@ pub const File = struct {
         };
     }
 
+    pub fn exists(allocator: &mem.Allocator, path: []const u8) bool {
```

Comment:
I think we should have this function be `access` rather than `exists` and on posix it should call the access system call. It can return error.FileNotFound as well as error.PermissionDenied.